crates/rover-client/src/operations/persisted_queries/publish/types.rs

```diff
@@ -0,0 +1,153 @@
+            "mutation" => Ok(Self::Mutation),
+            "subscription" => Ok(Self::Subscription),
+            input => Err(RoverClientError::AdhocError { msg: format!("'{input}' is not a valid operation type. Must be one of: 'query', 'mutation', or 'subscription'.") })
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for PersistedQueryOperationType {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        FromStr::from_str(&s).map_err(de::Error::custom)
+    }
+}
+
+impl Display for PersistedQueryOperationType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let result = match &self {
+            Self::Query => "QUERY",
+            Self::Mutation => "MUTATION",
+            Self::Subscription => "SUBSCRIPTION",
+        };
+
+        write!(f, "{result}")
+    }
+}
+
+impl From<PersistedQueriesPublishInput> for QueryVariables {
+    fn from(input: PersistedQueriesPublishInput) -> Self {
+        Self {
+            graph_id: input.graph_id,
+            list_id: input.list_id,
+            operation_manifest: Some(
+                input
+                    .operation_manifest
+                    .operations
+                    .iter()
+                    .cloned()
+                    .map(|operation| PersistedQueryInput {
+                        name: operation.name,
+                        body: operation.body,
+                        id: operation.id,
+                        type_: match operation.r#type {
+                            PersistedQueryOperationType::Mutation => OperationType::MUTATION,
+                            PersistedQueryOperationType::Subscription => {
+                                OperationType::SUBSCRIPTION
+                            }
+                            PersistedQueryOperationType::Query => OperationType::QUERY,
+                        },
+                    })
+                    .collect(),
+            ),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct PersistedQueriesPublishResponse {
+    pub revision: i64,
+    pub graph_id: String,
+    pub list_id: String,
+    pub result: PersistedQueriesPublishResponseType,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum PersistedQueriesPublishResponseType {
+    New(PersistedQueriesPublishResponseNewRevision),
+    Unchanged,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct PersistedQueriesPublishResponseNewRevision {
+    pub added: i64,
+    pub identical: i64,
+    pub removed: i64,
+    pub unaffected: i64,
+    pub updated: i64,
+}
+
+impl PersistedQueriesPublishResponseNewRevision {
+    pub fn added_str(&self) -> Option<String> {
+        Self::ops_str(self.added)
+    }
+
+    pub fn updated_str(&self) -> Option<String> {
+        Self::ops_str(self.updated)
+    }
+
+    pub fn removed_str(&self) -> Option<String> {
```

Comment:
FWIW removed should always be zero since you're not passing the remove argument to publishOperations.